Wikipedia/Code/WMFRelatedSectionBlackList.m

```diff
@@ -51,70 +60,20 @@ + (instancetype)loadFromDisk {
 
 - (void)addBlackListTitle:(MWKTitle*)title {
```

Comment:
now this class has two mutation APIs: one thread-safe and one not.  it would be simpler and less error-prone if `addEntry:`, `removeEntry:`, etc. were overridden to be thread safe, instead of creating a thread safe API on top of it.  you could even go so far as to make a `MWKAtomicList` subclass which does this, and have `WMFBlacklist` subclass that.